bcbio/variation/coverage.py

```diff
@@ -208,12 +207,17 @@ def run_mosdepth(data, target_name, bed_file, per_base=False, quantize=None, thr
     bam_file = dd.get_align_bam(data) or dd.get_work_bam(data)
     work_dir = utils.safe_makedir(os.path.join(dd.get_work_dir(data), "coverage", dd.get_sample_name(data)))
     prefix = os.path.join(work_dir, "%s-%s" % (dd.get_sample_name(data), target_name))
+    thresholds = thresholds and LooseVersion(check_output(["mosdepth", "--version"]).split()[1].strip()) >= LooseVersion("0.2.0")
     out = MosdepthCov("%s.mosdepth.dist.txt" % prefix,
                       ("%s.per-base.bed.gz" % prefix) if per_base else None,
                       ("%s.regions.bed.gz" % prefix) if bed_file else None,
                       ("%s.quantized.bed.gz" % prefix) if quantize else None,
                       ("%s.thresholds.bed.gz" % prefix) if thresholds else None)
-    if not utils.file_uptodate(out.dist, bam_file):
+    if not utils.file_uptodate(out.dist, bam_file) \
+            or (per_base and not utils.file_uptodate(out.per_base, bam_file)) \
+            or (bed_file and not utils.file_uptodate(out.regions, bam_file)) \
+            or (quantize and not utils.file_uptodate(out.quantize, bam_file)) \
+            or (thresholds and not utils.file_uptodate(out.thresholds, bam_file)):
```

Comment:
I'd skip all these extra checks here. If the user has run it previously we don't want to re-run, but later just skip if the thresholds file isn't present.